api/nakadi-event-bus-api.yaml

```diff
@@ -574,6 +574,25 @@ paths:
             The flow id of the request, which is written into the logs and passed to called services. Helpful
             for operational troubleshooting and log analysis.
           type: string
+        - name: X-nakadi-cursors
+          in: header
```

Comment:
Please transform this header into a query parameter or body. We have discussed this aspect thoroughly in the API guilde and we all share the consent, that headers should be reserved for contextual information like language or flow-ids that are not part of the data / data model of the API.

A significant request parameter should always be transported via GET as a query parameter (or as path parameter, if suitable as resource identifier) or via POST as a body content. We acknowledge the problem that servers implement size constraints for URLs, but this is in the same way true for headers. The length gains are very limited and depend more on the concrete server/router implementation/setup than on the aspect whether it is a query parameter or a header parameter. Therefore, we do not see a good reason for violating the basic conceptual primitives and to reserve headers for contextual information.

We know that you have used this pattern before, but base on the feedback of API consumers, we suggest to stop further spreading of this pattern and may even ask you to consider smoothly changing the design of the two existing places. We will shortly improve our API guidelines, to give better guidance on the usage of headers and get parameters with respect to semantic and size constraints.